sqlmesh/core/engine_adapter/clickhouse.py

```diff
@@ -0,0 +1,461 @@
+                table_name_or_schema.this
+                if isinstance(table_name_or_schema, exp.Schema)
+                else table_name_or_schema
+            )
+            self._insert_append_query(
+                table_name,
+                expression,  # type: ignore
+                columns_to_types or self.columns(table_name),
+            )
+
+    def _build_table_properties_exp(
+        self,
+        catalog_name: t.Optional[str] = None,
+        storage_format: t.Optional[str] = None,
+        partitioned_by: t.Optional[t.List[exp.Expression]] = None,
+        partitioned_by_user_cols: t.Optional[t.List[exp.Expression]] = None,
+        partition_interval_unit: t.Optional[IntervalUnit] = None,
+        clustered_by: t.Optional[t.List[str]] = None,
+        table_properties: t.Optional[t.Dict[str, exp.Expression]] = None,
+        columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
+        table_description: t.Optional[str] = None,
+        table_kind: t.Optional[str] = None,
+        empty_ctas: bool = False,
+    ) -> t.Optional[exp.Properties]:
+        properties: t.List[exp.Expression] = []
+
+        table_engine = self.DEFAULT_TABLE_ENGINE
+        if storage_format:
+            table_engine = (
+                storage_format.this if isinstance(storage_format, exp.Var) else storage_format  # type: ignore
+            )
+        properties.append(exp.EngineProperty(this=table_engine))
+
+        # copy of table_properties so we can pop items off below then consume the rest later
+        table_properties_copy = {
+            k.upper(): v for k, v in (table_properties.copy() if table_properties else {}).items()
+        }
+
+        if bool(re.search(self.ORDER_BY_TABLE_ENGINE_REGEX, table_engine)):
+            ordered_by_raw = table_properties_copy.pop("ORDER_BY", None)
+            ordered_by_cols = []
+            if ordered_by_raw:
+                # ORDER_BY: col
+                if isinstance(ordered_by_raw, exp.Column):
+                    ordered_by_cols.append(ordered_by_raw)
+                # ORDER_BY: 'col'
+                elif isinstance(ordered_by_raw, exp.Literal):
+                    ordered_by_cols.append(ordered_by_raw.this)
+                # ORDER_BY: (col1, col2) or ('col1', 'col2')
+                elif isinstance(ordered_by_raw, exp.Tuple):
+                    for col in ordered_by_raw.expressions:
+                        col = col.this if isinstance(col, exp.Literal) else col
+                        ordered_by_cols.append(col)
+                # python list of columns
+                elif isinstance(ordered_by_raw, list):
+                    ordered_by_cols.extend(ordered_by_raw)
+
+            ordered_by_expressions = (
+                exp.Tuple(expressions=[exp.to_column(k) for k in ordered_by_cols])  # type: ignore
+                if ordered_by_cols
+                # default () if no columns provided
+                else exp.Literal(this="()", is_string=False)
```

Comment:
Why are we constructing `()` like this? Can't we just create a `Tuple` with an empty expression list? So there's no need for a branch.